dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerAccessManager.java

```diff
@@ -584,6 +615,24 @@ public List<String> canRead(
     return errors;
   }
 
+  @Override
+  public List<String> canRead(
+      @Nonnull UserDetails user, SingleEvent event, DataElement dataElement) {
```

Comment:
Probably not part of this PR, but I thought we weren’t able to validate ACLs on data elements nor attributes, since we can't support that level of granularity.
I see this method is called only when the user requests a file. Is this the only accepted exception of this use case?
@ameenhere